### PR TITLE
ReadComicFree Source

### LINKED ISSUE #1571: Source Request: Read Comics Free
### Source name

Read Comics Free

### Source link

https://readcomic.me/

### Source language

English

### Other details

_No response_

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [extension listing](https://keiyoushi.github.io/extensions/) and verified it does not appear there.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by barrelltitor**
Closes #1571

Search path segment is a bit ugly, if anyone knows a better way to do it I'd be happy to change

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/en/readcomicfree/src/eu/kanade/tachiyomi/extension/en/readcomicfree/ReadComicFree.kt [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                    is StatusFilter -> {
+                        statusSegment = if (filter.toUriPart().isNullOrBlank()) "" else filter.toUriPart()
+                    }
+                    is SortFilter -> {
+                        sortSegment = if (filter.toUriPart().isNullOrBlank()) "" else filter.toUriPart()
+                    }
+                    else -> {}
+                }
+            }
+            // Incredibly ugly, but works
```
Comment (AwkwardPeak7) [inline src/en/readcomicfree/src/eu/kanade/tachiyomi/extension/en/readcomicfree/ReadComicFree.kt]: consider using shortcircuit logic to simplify this

File: src/en/readcomicfree/src/eu/kanade/tachiyomi/extension/en/readcomicfree/ReadComicFree.kt [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl/comic-updates".toHttpUrl().newBuilder().apply {
+            if (page > 1) addQueryParameter("page", page.toString())
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val urlBuilder = baseUrl.toHttpUrl().newBuilder()
```
Comment (AwkwardPeak7) [inline src/en/readcomicfree/src/eu/kanade/tachiyomi/extension/en/readcomicfree/ReadComicFree.kt]: just use one `.apply` and add everything in there, this way you directly call `addQueryParameter`

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

**Comment by barrelltitor** [comment]
Looks like the pages aren't loading now for some reason, checking to see if they changed their css already or if there's something wrong on my end, will reply once that is sorted out. I've made the other changes

**Comment by barrelltitor** [comment]
All fixed and working!

**Comment by AwkwardPeak7** [comment]
This too appears to be WpComics, use the multisrc

**Comment by AwkwardPeak7** [comment]
closing due to inactivity, feel free to open a new pull request

**Pull request closed without merging**